src/app/components/SignUp/SignUp.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { toast, ToastContainer } from 'react-toastify';
+import { z } from 'zod';
+
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Button } from '@lib/Button/Button';
+import { Checkbox } from '@lib/Checkbox/Checkbox';
+import { Form, FormControl, FormField, FormItem, FormMessage } from '@lib/Form/Form';
+import { Input } from '@lib/Input/Input';
+
+import 'react-toastify/dist/ReactToastify.css';
+
+const formSchema = z.object({
+  email: z.string().min(2, { message: 'Please enter your email address' }).max(50),
+  username: z.string().min(2, { message: 'Name must be atleast 2 characters' }).max(50),
+  password: z.string().min(2, { message: 'Password be atleast 2 characters' }).max(50),
+  confirmPassword: z.string().min(2, { message: 'Confirm password must be atleast 2 characters' }).max(50),
```

Comment:
- Add 'tos' into schema
- Add logic to say when it fails show the toast
- Remove the checkbox useState

Reply:
Updated